src/vs/workbench/api/common/extHostNotebookDocument.ts

```diff
@@ -294,6 +294,10 @@ export class ExtHostNotebookDocument {
 			}
 		}
 
+		// Freeze event properties so handlers cannot accidentally modify them
+		deepFreeze(result.cellChanges);
+		deepFreeze(result.contentChanges);
```

Comment:
shallow freeze only, otherwise `NotebookCell` instances are frozen

Reply:
Yes this should just be object.freeze instead of deepFreeze

Reply:
Although it's weird that the output wasn't frozen? I think I need to debug the actual outputs here to see what's in them. Object. freeze would certainly work but there might be more that can be frozen.